File: modia-syforest-provider/src/main/java/no/nav/sbl/dialogarena/modiasyforest/rest/domain/Arbeidsgiver.java
package no.nav.sbl.dialogarena.modiasyforest.rest.domain;

public class Arbeidsgiver {

    public String navn;
    public String orgnummer;

    public Arbeidsgiver withNavn(final String navn) {
        this.navn = navn;
        return this;
    }

    public Arbeidsgiver withOrgnummer(final String orgnummer) {
        this.orgnummer = orgnummer;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Arbeidsgiver that = (Arbeidsgiver) o;

        return orgnummer.equals(that.orgnummer) && navn != null ? navn.equals(that.navn) : that.navn == null;
    }

    @Override
    public int hashCode() {
        int result = navn != null ? navn.hashCode() : 0;
        result = 31 * result + orgnummer.hashCode();
        return result;
    }
}

File: src/main/kotlin/no/nav/syfo/controller/domain/Arbeidsgiver.kt
package no.nav.syfo.controller.domain

import java.io.Serializable

class Arbeidsgiver : Serializable {
    var navn: String? = null
    lateinit var orgnummer: String
    fun withNavn(navn: String?): Arbeidsgiver {
        this.navn = navn
        return this
    }

    fun withOrgnummer(orgnummer: String): Arbeidsgiver {
        this.orgnummer = orgnummer
        return this
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as Arbeidsgiver
        return if (orgnummer == that.orgnummer && navn != null) navn == that.navn else that.navn == null
    }

    override fun hashCode(): Int {
        var result = if (navn != null) navn.hashCode() else 0
        result = 31 * result + orgnummer.hashCode()
        return result
    }
}
